src/dodal/devices/oav/oav_to_redis_forwarder.py

```diff
@@ -79,20 +81,32 @@ async def _get_frame_and_put_to_redis(self, response: ClientResponse):
         self.uuid_setter(image_uuid := str(uuid.uuid4()))
         img = Image.open(io.BytesIO(jpeg_bytes))
         image_data = pickle.dumps(np.asarray(img))
-        await self.redis_client.hset(self.redis_key, image_uuid, image_data)  # type: ignore
-        LOGGER.debug(f"Sent frame to redis key {self.redis_key} with uuid {image_uuid}")
+        sample_id = str(await self.sample_id.get_value())
+        await self.redis_client.hset(sample_id, image_uuid, image_data)  # type: ignore
+        await self.redis_client.expire(sample_id, timedelta(days=self.DATA_EXPIRY_DAYS))
+        LOGGER.debug(f"Sent frame to redis key {sample_id} with uuid {image_uuid}")
 
-    async def _stream_to_redis(self):
+    async def _open_connection_and_(self, function_to_do: Callable):
```

Comment:
Nit: Function name looks a bit weird to me, maybe `_open_connection_and_do_function`?